### ISSUE TITLE
Problem with the Sandy test case (WRF-SWAN): invalid boundary point for SWAN

### ISSUE THREAD

**Issue by nothing-ends-adrian** [DOI-USGS/COAWST#48]
Hey John, 

I'm looking to run WRF-SWAN for my application and am starting out with the Sandy test case (coawst v3.6). 
Compilation was fine (gnerated coawstM, didn't seem to throw any errors in the build), but I keep hitting a SWAN boundary issue when running the test case.

For example in one of my PRINT outputs I get - 

` BOUNDSPEC SEGMENT XY -64.7287 27.6450 -64.2858 30.1019 VARIABLE FILE 0 'TPAR1.txt'
 ** Error            : invalid boundary point
 segment point     -64.73     27.65 grid   -99.00  -99.00 -98 -98
 ** Error            : (  -99  -99) is outside computational grid
 ** Error            : invalid boundary point
 segment point     -64.29     30.10 grid   -99.00  -99.00 -98 -98
 ** Error            : (  -99  -99) is outside computational grid
 ** Warning          : No points on the boundaries found
 ** Warning          : At least two points needed for a segment
 ** Warning          : Length of segment short, boundary values ignored
 segment length=     0.00; [len]=     0.00`

I'm really perplexed about this error as I'm just using the same inputs as in the Sandy test case (nothing user generated by me or specific). Furthermore, the BOUNDSPEC is -64.7287 27.6450 -64.2858 30.1019 so I have no idea where SWAN getting -99.00  -99.00?
I'm running the WRF-SWAN (Sandy) with 48 cores, (24 to SWAN, 24 to WRF).

Thank you for your time! 
Luke

**Comment by jcwarner-usgs** [comment]
a couple of things to try.
sometimes SWAN has a roundoff error when it tries to compute if the point is inside the grid. as a test, try using the SEGMENT IJ lines instead of the SEGMENT XY lines.

also, 24 processors might be too much for this very simple case. try it with like 10.

-j

**Comment by nothing-ends-adrian** [comment]
I switched to IJ and it's running! 

However, I couldn't figure out how to convert the sandy test case XY boundary coords to IJ coords so I ended up putting in a constant boundary, similar to the JOE_TC case, just to see if it will start running.

`& Boundary files  ****************************************
BOUND SHAPESPEC JONSWAP PEAK DSPR DEGREES
BOUNDSPEC SEGMENT IJ   0   0 86   0 CONSTANT PAR 2.0 14.0 180. 20.
BOUNDSPEC SEGMENT IJ 86   0 86 64 CONSTANT PAR 2.0 14.0 90. 20.
BOUNDSPEC SEGMENT IJ   0 64 86 64 CONSTANT PAR 2.0 14.0 0. 20.
BOUNDSPEC SEGMENT IJ  0   0   0 64 CONSTANT PAR 2.0 14.0 270. 20.`

Now I'm going to move onto my own application. However, ww3_swan_input.m and bdry_com.m only produces BOUNDSPEC SEGMENT XY TPAR files. Any ideas on how I can get it running with XY or easily convert XY to IJ? 

Thanks!

**Comment by jcwarner-usgs** [comment]
are you good at trying things?
the specres array has the indices in it, they are in colums 5-8.  i use columns 1-4 to write out the XY locations, but you could change the code to write out the IJ locations from columns 5-8.  give something like this a try:

edit bdry_com.m
and change line 16 from
   fprintf(fid,'%3.4f %3.4f %3.4f %3.4f ',specpts(bd,1:4));
to
   fprintf(fid,'%3.4f %3.4f %3.4f %3.4f ',specpts(bd,5:8));

and see what that does.
also notice, we updated the readww3_2TPAR routine itself, but the specres stuff is still the same.

**Comment by jcwarner-usgs** [comment]
you would also need to change the
   fprintf(fid,'BOUNDSPEC SEGMENT XY '); 
to
   fprintf(fid,'BOUNDSPEC SEGMENT IJ ');

**Comment by nothing-ends-adrian** [comment]
Look like this has done the trick! Thank you John :)

I ended up grabbing the latest programs in coast 3.6 (create_swanTpar_from_WW3.m and modifying that bdry_com.m). I also had to convert my grb2 files to nc files (I think this may have been an issue with my rather old matlab version 2017). 

One last Q (slightly off topic). How do I run through several months in one simulation?

From the looks of it the TPAR file generation matlab program, it only gives the boundary for a month at a time. 
So can run say create_swanTpar_from_WW3.m (updated version of ww3_swan_input.m) for each month (say three months) and then I just CAT all these months together? So a TPAR1 file could go across different months like so - 

20170630.2100         2.01        14.90     207.       20
20170701.0000         1.59        9.60     121.       20

Then for my swan*.in file for the INPgrid I put something like 90 DAY (for a 3 month simulation) - 

&& KEYWORD TO CREATE CURRENT GRID &&
INPGRID CURRENT CURVILINEAR 0 0 283 253 EXC 9.999000e+003 &
       NONSTATIONARY 20170601.000000 90 DAY 20170901.000000

&& KEYWORD TO CREATE WATER LEVEL GRID &&
INPGRID WLEV CURVILINEAR 0 0 283 253 EXC 9.999000e+003 &
       NONSTATIONARY 20170601.000000 90 DAY 20170901.000000

&& KEYWORD TO CREATE BOTTOM FRIC GRID &&
INPGRID FRIC CURVILINEAR 0 0 283 253 EXC 9.999000e+003 &
       NONSTATIONARY 20170601.000000 90 DAY 20170901.000000

Thanks,
Luke
